packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AbstractModal from 'terra-abstract-modal';
+import Button from 'terra-button';
```

Comment:
This is a little tricky, the Notification Dialog itself doesn't actually use Buttons, it only needs the type check. I'm not sure if this component gains enough value from the type check to justify the hard Button dependency, potential package size increase, and multiple Button versions. Does anyone else have any thoughts on this?

Reply:
One other option is to change the type checking to `componentWithName('Button')` .
This would allow any component with name Button and is not restricted to `terra-button`. Obviously, it's not foolproof though( anyone could rename any component to button ).

Reply:
If we only want to accept Buttons here, would it make more sense to have the notification dialog render the buttons itself? Resulting in an API like:

```jsx
<NotificationDialog
  actions={[
    { text: 'OK', onClick: () => {} },
    { text: 'Cancel', onClick: () => {} }
  ],
  ...
/>
```

You could alternatively tie keys to each action and have a single selection function to filter them:
```jsx
<NotificationDialog
  actions={[
    { key: 'ok', text: 'OK' },
    { key: 'cancel', text: 'Cancel'}
  ],
  onSelect={(actionKey) => {
    if (actionKey === 'cancel') {
      // do something
    } else if (actionKey === 'ok') {
      // do something else
    }
  }}
/>
```
You may need more of the Button props exposed, but if we're really wanting to limit the actions to Buttons, it may be worth considering.

Reply:
This sounds like an interesting solution. But till what extent do we want to restrict the keys?

Reply:
Changed this to restrict using only two buttons.

Reply:
@neilpfeiffer Do we want to restrict the footer buttons this much? I see the value in controlling which button has emphasis, but it seems like a 3+ button footer is going to come up sooner or later.

Reply:
If we do only want two buttons, we could just expose two separate props to make that really clear:

```
<NotificationDialog
  primaryAction={{
    text: 'OK',
    onClick: () => {},
  }}
  secondaryAction={{
    text: 'Cancel',
    onClick: () => {},
  }}
>
```

Reply:
Thanks @tbiethman - @rakshith91 and I already discussed and I like taking the approach of more restrictive first (only 2 buttons) and only expand when we get requests (following Cory House's recommendation of being rigid to start https://youtu.be/MsTARVOl1H4?t=15m26s ).  

Current terra-alert (Notification Banners) restricts to only one custom action plus a dismiss, so it is safe to follow the same paradigm here to start as well as best practice, since the two are sister components with similar usage.

Reply:
Changed the API to pass `primaryAction` and `SecondaryAction` instead.